app/models/agent.rb

```diff
@@ -61,8 +61,8 @@ class Agent < ActiveRecord::Base
   has_many :scenario_memberships, :dependent => :destroy, :inverse_of => :agent
   has_many :scenarios, :through => :scenario_memberships, :inverse_of => :agents
 
-  scope :active,   -> { where(disabled: false) }
-  scope :inactive, -> { where(disabled: true) }
+  scope :active,   -> { joins(:user).where('users.deactivated_at IS NULL').where(disabled: false) }
+  scope :inactive, -> { joins(:user).where(["users.deactivated_at IS NOT NULL OR disabled = ?", true]) }
```

Comment:
I'd suggest instead that agents get a new field like `deactivated_by_system` that gets set to true when the user is deactivated. There'd also need to be a check that no new agents could be created by deactivated users. But by adding a new column to agents, you prevent the tight coupling to users, and open agents up to be deactivated for other reasons by the system.